File: 462-minimum-moves-to-equal-array-elements-ii/minimum-moves-to-equal-array-elements-ii.cpp
class Solution {
public:
    int minMoves2(vector<int>& nums) {
        sort(begin(nums),end(nums));
        int idx=-1;
        int cnt=0;
        if(nums.size()%2!=0) {
            idx=nums.size()/2;
            int val=nums[idx];
            for(int i=0; i<nums.size(); i++) {
                cnt+=abs(val-nums[i]);
            }
            
        }
        else {
            idx=nums.size()/2;
            int idx2=idx-1;
            int temp1=0;
            int temp2=0;
            for(int i=0; i<nums.size(); i++) {
                temp1+=abs(nums[idx2]-nums[i]);
                temp2+=abs(nums[idx]-nums[i]);
            }
            return min(temp1,temp2);
        }
          return cnt;
    }
};
//1 2 3 4 5 6 

File: 462-minimum-moves-to-equal-array-elements-ii/minimum-moves-to-equal-array-elements-ii.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minMoves2 = function(nums) {
        nums.sort((a,b) => a-b);
        let cnt=0;
let n=nums.length;
        if(n%2 !==0) {
            let idx=Math.floor(n/2);
            for(let i=0; i<n; i++) {
                cnt+= Math.abs(nums[i]-nums[idx]);
            }
        }
        else {
            let idx=Math.floor(n/2);
            let idx2=idx-1;
            let temp1=0;
            let temp2=0;
            for(let i=0; i<n; i++) {
              temp1+=Math.abs(nums[i]-nums[idx]);
              temp2+=Math.abs(nums[i]-nums[idx2]);
            }
            return Math.min(temp1,temp2);        }
  return cnt;

};